utility_analysis_new/combiners.py

```diff
@@ -478,3 +475,186 @@ def metrics_names(self) -> List[str]:
+                               metrics.std_noise**2)
+        abs_error_quantiles = []
+        for quantile in self._error_quantiles:
+            error_at_quantile = scipy.stats.norm.ppf(
+                q=1 - quantile, loc=loc_cpe_ne,
+                scale=std_cpe_ne) + metrics.per_partition_error
+            abs_error_quantiles.append(error_at_quantile)
+
+        # Relative error metrics
+        if metrics.count == 0:  # For empty public partitions, to avoid division by 0
+            rel_error_expected = float('inf')
+            rel_error_variance = float('inf')
+            rel_error_quantiles = [float('inf')] * len(self._error_quantiles)
+        else:
+            rel_error_expected = abs_error_expected / metrics.count
+            rel_error_variance = abs_error_variance / (metrics.count**2)
+            rel_error_quantiles = [
+                error / metrics.count for error in abs_error_quantiles
+            ]
+
+        return AggregateErrorMetricsAccumulator(
+            num_partitions=1,
+            abs_error_expected_sum=abs_error_expected,
+            abs_error_variance_sum=abs_error_variance,
+            abs_error_quantiles_sum=abs_error_quantiles,
+            rel_error_expected_sum=rel_error_expected,
+            rel_error_variance_sum=rel_error_variance,
+            rel_error_quantiles_sum=rel_error_quantiles,
+        )
+
+    def merge_accumulators(self, acc1: AccumulatorType, acc2: AccumulatorType):
+        """Merges two accumulators together additively."""
+        return acc1 + acc2
+
+    def compute_metrics(self, acc: AccumulatorType) -> AggregateErrorMetrics:
+        """Compute metrics based on the accumulator properties."""
+        return AggregateErrorMetrics(
+            abs_error_expected_avg=acc.abs_error_expected_sum /
+            acc.num_partitions,
+            abs_error_variance_avg=acc.abs_error_variance_sum /
+            acc.num_partitions,
+            abs_error_quantiles_avg=[
+                sum / acc.num_partitions for sum in acc.abs_error_quantiles_sum
+            ],
+            rel_error_expected_avg=acc.rel_error_expected_sum /
+            acc.num_partitions,
+            rel_error_variance_avg=acc.rel_error_variance_sum /
+            acc.num_partitions,
+            rel_error_quantiles_avg=[
+                sum / acc.num_partitions for sum in acc.rel_error_quantiles_sum
+            ])
+
+    def metrics_names(self) -> List[str]:
+        return [
+            'abs_error_expected_avg', 'abs_error_variance_avg',
+            'abs_error_99pct_avg', 'rel_error_expected_avg',
+            'rel_error_variance_avg', 'rel_error_99pct_avg'
+        ]
+
+    def explain_computation(self):
+        pass
+
+
+class PrivatePartitionSelectionAggregateErrorMetricsCombiner(
+        pipeline_dp.Combiner):
+    """A combiner for aggregating errors across partitions for private partition selection"""
+    # TODO: Implement logic.
```

Comment:
Should we drop this todo?

Reply:
This will be implemented in a follow-up PR, the current version is just a placeholder and doesn't produce anything meaningful.